### ISSUE TITLE
Show thumbnails in 'file_list' as squares

### ISSUE THREAD

**Issue by fakeartist** [WebDevStudios/CMB2#625]
Is there anyway, so that I can show the thumbnails in 'file_list' as squares? In other words, the same way as a default gallery in wordpress works? Or in a way that the cropped thumbnails are seen?

**Comment by arkonisus** [comment]
It wold be essential if you try to add different file-types e.g. images, pdf, zip. Per default you will see an image next to a filename and so on ... 

I agree to @fakeartist - the best format wuld be a square or the support of the parameter 'preview_size' also for non-image-files.

**Comment by jtsternberg** [comment]
You can specify any size preview you want: https://github.com/WebDevStudios/CMB2/wiki/Field-Types#custom-field-attributes-5

This parameter can also take a named thumbnail size, so you can use any image size you have registered with [`add_image_size()`](https://developer.wordpress.org/reference/functions/add_image_size/).

As for the 2nd request, "the best format would be a square or the support of the parameter 'preview_size' also for non-image-files", I'm not entirely sure how you expect that to work. There is no way for CMB2 to know what file types you have allowed on your site, and/or how to display them as a preview with a specific size.

**Comment by arkonisus** [comment]
@jtsternberg Yes, for images the size-attribute is ok, but if you try to attach images and e.g. PDFs to the file_list you are not able to sort the items because of the different size. 

If the items would look similar to the standard media-view as you can see in the screenshot the sorting is not a problem. As a workaround it is possible to style the list-tags in the admin screen, but it would be much better if the default view of the file_list could handle it in a better way.

You can also take a look at the second attachment. This is the file_list feature of pods.io. There you can see the same icons as you see in the media-view. So i think there must be a solution ... :)

![media](https://cloud.githubusercontent.com/assets/7242966/14398830/88054a2e-fde7-11e5-8141-9640602c365d.png)
![pods](https://cloud.githubusercontent.com/assets/7242966/14398836/9371efde-fde7-11e5-980d-24ce3c6d88a6.png)

**Comment by jtsternberg** [comment]
Ah, that makes a ton of sense, and thank you for the screenshots. I don't think that will be difficult to pull off.

**Comment by fakeartist** [comment]
The default size of the preview images should be 50x50 (cropped). In my case, if I add portrait and landscape images, look what happens in the attachment. Seems like a bug to me.
![screenshot_2](https://cloud.githubusercontent.com/assets/2536468/14403029/eb1bd94e-fe4f-11e5-8b42-72521b17c3c5.jpg)
